src/tools/lld-wrapper/src/main.rs

```diff
@@ -0,0 +1,140 @@
+//! Script to invoke the bundled rust-lld with the correct flavor based on the executable name.
+//!
+//! lld supports multiple command line interfaces. If `-flavor <flavor>` are passed as the first
+//! two arguments the `<flavor>` command line interface is used to process the remaining arguments.
+//! If no `-flavor` argument is present the flavor is determined by the executable name.
+//!
+//! In Rust with `-Z gcc-ld=lld` we have gcc or clang invoke rust-lld. Since there is no way to
+//! make gcc/clang pass `-flavor <flavor>` as the first two arguments in the linker invocation
+//! and since Windows does not support symbolic links for files this wrapper is used in place of a
+//! symblic link. It execs `../rust-lld -flavor ld` if its name is `ld` and `../rust-lld -flavor ld64`
+//! if it is named `ld64`. On Windows it spawns a `..\rust-lld.exe` child process.
```

Comment:
Instead of trying to use the name of the executable to detect this, can we just build it twice (e.g., with a cargo feature for each variant)? It seems like that would both be simpler and easier to understand.

Reply:
Indeed.